src/components/map/use-map-tariff-zones.ts

```diff
@@ -0,0 +1,112 @@
+import { useCallback, useEffect, useState } from 'react';
+import { Language, useTranslation } from '@atb/translations';
+import mapboxgl from 'mapbox-gl';
+import { getReferenceDataName } from '@atb/utils/reference-data';
+import { centroid } from '@turf/turf';
+import { type TariffZone, getTariffZones } from '@atb/modules/firebase';
+
+export const useMapTariffZones = async (
+  mapRef: React.MutableRefObject<mapboxgl.Map | undefined>,
+) => {
+  const { language } = useTranslation();
+  const [tariffZones, setTariffZones] = useState<TariffZone[]>([]);
+
+  useEffect(() => {
+    const fetchTariffZones = async () => {
+      const zones = await getTariffZones();
+      setTariffZones(zones);
+    };
+
+    fetchTariffZones();
+  }, []);
+
+  const addTariffZonesToMap = useCallback(
+    (map: mapboxgl.Map) => {
+      if (!map.getSource('tariff-zones')) {
+        map.addSource(
+          'tariff-zones',
+          createTariffZonesFeatureCollection(tariffZones, language),
+        );
+      }
+
+      if (!map.getLayer('zone-boundary-layer')) {
+        map.addLayer({
+          id: 'zone-boundary-layer',
+          type: 'line',
+          source: 'tariff-zones',
+          paint: {
+            'line-width': 1,
+            'line-color': '#666666',
+          },
+        });
+      }
+
+      if (!map.getLayer('zone-names-layer')) {
+        map.addLayer({
+          id: 'zone-names-layer',
+          type: 'symbol',
+          source: 'tariff-zones',
+          layout: {
+            'text-field': ['get', 'name'],
+            'text-justify': 'auto',
+            'text-offset': [0, 0],
+          },
+          paint: {
+            'text-color': '#000000',
+            'text-halo-color': '#FFFFFF',
+            'text-halo-width': 2,
+          },
+          filter: ['==', '$type', 'Point'],
+        });
+      }
+    },
+    [language, tariffZones],
+  );
+
+  useEffect(() => {
+    if (!mapRef || !mapRef.current || !tariffZones.length) return;
+    const map = mapRef.current;
+
+    if (map.loaded()) {
+      addTariffZonesToMap(map);
+    }
+
+    map.on('load', () => {
+      addTariffZonesToMap(map);
+    });
+  }, [mapRef, tariffZones, language, addTariffZonesToMap]);
+};
+
+const createTariffZonesFeatureCollection = (
+  tariffZones: TariffZone[],
+  language: Language,
+) => ({
+  type: 'geojson' as const,
+  data: {
+    type: 'FeatureCollection' as const,
+    features: tariffZones
+      .map((tariffZone) => [
+        {
+          type: 'Feature' as const,
+          properties: {},
+          geometry: {
+            type: 'Polygon' as const,
+            coordinates: tariffZone.geometry.coordinates,
+          },
+        },
+        {
+          type: 'Feature' as const,
+          properties: {
+            name: getReferenceDataName(tariffZone, language),
```

Comment:
Would it be a good idea to prefix the zone names with "Zone"? E.g. "Zone Molde" so that we are more clear in what's shown on the map